lib/src/main/java/org/altbeacon/beacon/startup/RegionBootstrap.java

```diff
@@ -34,24 +35,106 @@
+        if (context.getApplicationContext() == null) {
+            throw new NullPointerException("Application Context should not be null");
+        }
+        beaconManager = BeaconManager.getInstanceForApplication(context.getApplicationContext());
+        this.context = context;
+        this.monitorNotifier = monitorNotifier;
+
+        this.monitorNotifier = new BootstrapNotifier() {
+            public Context getApplicationContext() {
+                return context.getApplicationContext();
+            }
+
+            public void didEnterRegion(Region region) {
+                monitorNotifier.didEnterRegion(region);
+            }
+
+            public void didExitRegion(Region region) {
+                monitorNotifier.didExitRegion(region);
+            }
+
+            public void didDetermineStateForRegion(int state, Region region) {
+                monitorNotifier.didDetermineStateForRegion(state, region);
+            }
+        };
+
+        regions = new ArrayList<Region>();
+        regions.add(region);
+        beaconConsumer = new InternalBeaconConsumer();
+        beaconManager.bind(beaconConsumer);
+        LogManager.d(TAG, "Waiting for BeaconService connection");
+    }
+
+    /**
+     * Constructor to bootstrap your Application on an entry/exit from multiple regions
+     *
+     * @param context
+     * @param regions
+     */
+    public RegionBootstrap(final Context context, final MonitorNotifier monitorNotifier, List<Region> regions) {
+        if (context.getApplicationContext() == null) {
+            throw new NullPointerException("Application Context should not be null");
+        }
+        beaconManager = BeaconManager.getInstanceForApplication(context.getApplicationContext());
+        this.context = context;
+        this.monitorNotifier = monitorNotifier;
+
+        this.monitorNotifier = new BootstrapNotifier() {
+            public Context getApplicationContext() {
+                return context.getApplicationContext();
+            }
+
+            public void didEnterRegion(Region region) {
+                monitorNotifier.didEnterRegion(region);
+            }
+
+            public void didExitRegion(Region region) {
+                monitorNotifier.didExitRegion(region);
+            }
+
+            public void didDetermineStateForRegion(int state, Region region) {
+                monitorNotifier.didDetermineStateForRegion(state, region);
+            }
+        };
+
+        this.regions = regions;
+
+        beaconConsumer = new InternalBeaconConsumer();
+        beaconManager.bind(beaconConsumer);
+        LogManager.d(TAG, "Waiting for BeaconService connection");
+    }
+
     /**
      * Constructor to bootstrap your Application on an entry/exit from a single region.
      *
      * @param application
      * @param region
+     * @deprecated This will be removed in a later release. Use
```

Comment:
Is there any reason to deprecate this?  It seems like it is a simpler way of using RegionBootstrap if you are using the Application class as the place to get your callbacks.

Reply:
Just thought that it may cause confusion but yes why not use BootstrapNotifier if you only have Application class.